subsys/bluetooth/host/smp.c

```diff
@@ -3052,13 +3052,12 @@ static int smp_send_pairing_req(struct bt_conn *conn)
 	req->auth_req = get_auth(smp, BT_SMP_AUTH_DEFAULT);
 	req->io_capability = get_io_capa(smp);
 
-	if (req->auth_req & BT_SMP_AUTH_SC) {
-		req->oob_flag = sc_oobd_present ? BT_SMP_OOB_PRESENT :
-				BT_SMP_OOB_NOT_PRESENT;
-	} else {
-		req->oob_flag = legacy_oobd_present ? BT_SMP_OOB_PRESENT :
-				BT_SMP_OOB_NOT_PRESENT;
-	}
+	/* At this point is it unknown if pairing will be legacy or LE SC so
+	 * set OOB flag if any OOB data is present and assume to peer device
+	 * provides OOB data that will match it's pairing type.
+	 */
+	req->oob_flag = (legacy_oobd_present || sc_oobd_present) ?
```

Comment:
Hi,

from my understanding, this fixes a problem but seems to bring a new one.
In the following scenario:
- central supports SC
- central supports OOB, but only legacy
- peripheral supports SC

central makes pair request with SC and OOB flags set, and because peripheral supports SC as well, the decided pairing method will be LE SC OOB. But central doesn't support LE SC OOB pairing, just legacy OOB

Reply:
My opinion is that if device supports LE SC and OOB  it should support LE SC OOB (since there is no way to differentiate legacy and SMP OOB data in SMP), otherwise IMHO it is device misconfiguration...

Reply:
The ideal case would be for a central to support both legacy and LE SC OOB. Maybe someone knows of a specification saying if it's required to also support legacy OOB if LE SC OOB is supported. In this case, supporting only legacy but not LE SC OOB would be a misconfiguration as you said